packages/vue-renderer/src/renderer.js

```diff
@@ -104,25 +104,34 @@ export default class VueRenderer {
   }
 
   async ready() {
-    if (!this.context.options.dev) {
-      // Production: Load SSR resources from fs
-      await this.loadResources(fs)
-
-      // Verify
-      if (this.context.options._start) {
-        if (!this.isReady) {
-          throw new Error(
-            'No build files found. Use either `nuxt build` or `builder.build()` or start nuxt in development mode.'
-          )
-        } else if (this.context.options.modern && !this.context.resources.modernManifest) {
-          throw new Error(
-            'No modern build files found. Use either `nuxt build --modern` or `modern` option to build modern files.'
-          )
-        }
-      }
-    } else {
-      // Development: Listen on build:resources hook
+    // -- Development mode --
+
+    if (this.context.options.dev) {
       this.context.nuxt.hook('build:resources', mfs => this.loadResources(mfs, true))
+      return
+    }
+
+    // -- Production mode --
+
+    // Try once to load SSR resources from fs
+    await this.loadResources(fs)
+
+    // Without using`nuxt start` (Programatic, Tests and Generate)
+    if (!this.context.options._start) {
+      this.context.nuxt.hook('build:resources', () => this.loadResources(fs))
```

Comment:
Should this line be await ?

Reply:
Returning a promise is same as awaiting on it @clarkdo :) (For the left side, `this.context` no await is required as we just register hook)

Reply:
I treat it as callHook😅